spec/API_specification/array_api/elementwise_functions.py

```diff
@@ -785,6 +785,21 @@ def greater_equal(x1: array, x2: array, /) -> array:
         an array containing the element-wise results. The returned array must have a data type of ``bool``.
     """
 
+def imag(x: array, /) -> array:
+    """
+    Returns the imaginary component of a complex number for each element ``x_i`` of the input array ``x``.
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a complex floating-point data type.
+
+    Returns
+    -------
+    out: array
+        an array containing the element-wise results. The returned array must have a floating-point data type with the same floating-point precision as ``x`` (e.g., if ``x`` is ``complex64``, the returned array must have the floating-point data type ``float32``). 
```

Comment:
Should this be "must" or "should"?

Reply:
In what scenarios should/would this not be the case? And why?

Reply:
Oh my bad I was reading this as if it were the input array, not the output. I see the input does say "should".

Reply:
For the input dtype, allowing array libraries to support real-valued dtypes makes sense in order to allow for backward compat; however, all array libraries should return an array having a dtype with the same precision.

I suppose you may be thinking if an array library supports real-valued int dtypes, would requiring a float dtype make sense.